modules/web/src/app/cluster/details/cluster/rbac/service-account/component.ts

```diff
@@ -0,0 +1,198 @@
+  bindingDetails: Record<string, SimpleClusterBinding[]>;
+  clusterServiceAccountExpansion: Record<string, boolean> = {};
+
+  @Input() cluster: Cluster;
+  @Input() projectID: string;
+  @Output() addBindingToServiceAccount = new EventEmitter<ClusterServiceAccount>();
+  @Output() deleteBinding = new EventEmitter<SimpleClusterBinding>();
+
+  constructor(
+    private readonly _rbacService: RBACService,
+    private readonly _matDialog: MatDialog,
+    private readonly _notificationService: NotificationService,
+    private readonly _clusterServiceAccountService: ClusterServiceAccountService
+  ) {}
+
+  ngOnInit(): void {
+    this._getServiceAccounts();
+    this._getBindings();
+  }
+
+  ngOnDestroy(): void {
+    this._unsubscribe.next();
+    this._unsubscribe.complete();
+  }
+
+  expandRow({id}: ClusterServiceAccount): void {
+    this.clusterServiceAccountExpansion[id] = !this.clusterServiceAccountExpansion[id];
+  }
+
+  addBinding(serviceAccount: ClusterServiceAccount): void {
+    this.addBindingToServiceAccount.emit(serviceAccount);
+  }
+
+  download(serviceAccount: ClusterServiceAccount): void {
+    window.open(
+      this._clusterServiceAccountService.getKubeconfigUrl(
+        this.projectID,
+        this.cluster.id,
+        serviceAccount.namespace,
+        serviceAccount.id
+      ),
+      '_blank'
+    );
+  }
+
+  deleteServiceAccount(element: ClusterServiceAccount): void {
+    const dialogConfig: MatDialogConfig = {
+      data: {
+        title: 'Delete Service Account',
+        message: `Delete service account <b>${element.name}</b> of <b>${this.cluster.name}</b> cluster permanently?`,
+        confirmLabel: 'Delete',
+      },
+    };
+
+    const dialogRef = this._matDialog.open(ConfirmationDialogComponent, dialogConfig);
+
+    dialogRef
+      .afterClosed()
+      .pipe(
+        filter(isConfirmed => isConfirmed),
+        switchMap(_ =>
+          this._clusterServiceAccountService.delete(this.projectID, this.cluster.id, element.namespace, element.name)
+        ),
+        take(1)
+      )
+      .subscribe(() => {
+        this._updateBindings();
+        this._notificationService.success(`Removed service account ${element.name} from ${this.cluster.name}`);
+      });
+  }
+
+  deleteServiceAccountBinding(element: SimpleClusterBinding): void {
+    this.deleteBinding.emit(element);
+  }
+
+  private _updateBindings(): void {
+    this._clusterServiceAccountService.update();
+    this._rbacService.refreshClusterBindings();
+    this._rbacService.refreshNamespaceBindings();
+  }
+
+  private _getServiceAccounts(): void {
+    this._clusterServiceAccountService
+      .get(this.projectID, this.cluster.id)
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(serviceAccount => {
+        this.dataSource.data = serviceAccount;
+        this.clusterServiceAccountExpansion = serviceAccount.reduce(
+          (prev: Record<string, boolean>, {id}) => ({
+            ...prev,
+            [id]: false,
```

Comment:
I think we should not do this because every time service accounts are refreshed it collapses the service account details.


https://user-images.githubusercontent.com/13975988/210236836-3e0fc930-2499-4f61-b763-76e4fb43b540.mov